python/cuml/preprocessing/__init__.py

```diff
@@ -13,7 +13,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-from cuml.preprocessing.model_selection import train_test_split
```

Comment:
If we want to remove this exposure, we need to go through a deprecation process. This will break at least one of our demos, so I recommend splitting it off into a separate PR.

Reply:
would it be appropriate to replace this with `from cuml.model_selection import train_test_split` for now?

Reply:
I think in general it's better to have `__init__` files import from the base source file rather than going through an additional layer of indirection.

Reply:
that makes sense, but is there a way to do this without duplicating code? If we do not move `model_selection.py` out from `preprocessing`, `train_test_split` could be imported from both `cuml.preprocessing.model_selection` and `cuml.model_selection` which may not be desirable

Reply:
We'll want it to be importable from both locations until we've gone through a deprecation process. We might start the deprecation process with this PR, though. We'll want a warning if users try to import from the old location, and then after a release cycle, we can eliminate it entirely.

Reply:
We should also get our examples updated accordingly.